src/sflow_plugin/sflow_collector.cpp

```diff
@@ -535,8 +542,27 @@ void decode_link_layer(SFSample* sample) {
     type_len = (ptr[0] << 8) + ptr[1];
     ptr += 2;
 
+    if (type_len == sflow_qinq_ethertype && ((ptr[2] << 8) + ptr[3]) == 0x8100) {
+        /* Outer VLAN tag - next two bytes */
+        uint32_t vlanData = (ptr[0] << 8) + ptr[1];
+        uint32_t vlan = vlanData & 0x0fff;
+        uint32_t priority = vlanData >> 13;
+        ptr += 2;
+
+        /*  _____________________________________ */
+        /* |   pri  | c |         vlan-id        | */
+        /*  ------------------------------------- */
```

Comment:
This diagram does not match reality :) Feel free to remove it.

Reply:
removed